src/codewhisperer/service/featureConfigProvider.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { FeatureValue } from '../client/codewhispereruserclient'
+import { codeWhispererClient as client } from '../client/codewhisperer'
+import { AuthUtil } from '../util/authUtil'
+import { getLogger } from '../../shared/logger'
+
+export class FeatureContext {
+    constructor(public name: string, public variation: string, public value: FeatureValue) {}
+}
+
+const testFeatureName = 'testFeature'
+const featureConfigPollIntervalInMs = 30 * 60 * 1000 // 30 mins
```

Comment:
@andrewyuq does this mean the featureConfig telemetry is emitted every 30 min? If so, why? Can it be triggered by a change event instead?